src/projects/repeat_resolution/mdbg.cpp

```diff
@@ -0,0 +1,838 @@
+        for (const RRVertexType &vertex : *this) {
+            auto[begin, end] = out_neighbors(vertex);
+            for (auto it = begin; it!=end; ++it) {
+                const RREdgeProperty &edge_prop = it->second.prop();
+                if (edge_prop.Size() > 0) {
+                    seq_edge.emplace(
+                        GetEdgeSequence(find(vertex), it, false, false)
+                            .ToSequence());
+                }
+            }
+        }
+
+        for (const RRVertexType &vertex : *this) {
+            auto[begin, end] = out_neighbors(vertex);
+            for (auto it = begin; it!=end; ++it) {
+                const RREdgeProperty &edge_prop = it->second.prop();
+                if (edge_prop.Size() > 0) {
+                    const Sequence seq =
+                        GetEdgeSequence(find(vertex), it, false, false)
+                            .ToSequence();
+                    VERIFY(seq_edge.count(seq)==1);
+                }
+            }
+        }
+    }
+}
+
+void MultiplexDBG::SpreadFrost() {
+    std::unordered_set<RRVertexType> prev_frozen, new_frozen;
+    for (const RRVertexType &vertex : *this) {
+        const RRVertexProperty &vertex_prop = node_prop(vertex);
+        if (vertex_prop.IsFrozen()) {
+            prev_frozen.insert(vertex);
+        }
+    }
+
+    auto upd_new_frozen = [this, &new_frozen](const RRVertexType &vertex,
+                                              const RRVertexProperty &vertex_prop,
+                                              NeighborsIterator begin,
+                                              NeighborsIterator end) {
+      for (auto it = begin; it!=end; ++it) {
+          const RREdgeProperty &edge_prop = it->second.prop();
+          const RRVertexType &neighbor = it->first;
+          const RRVertexProperty &neighbor_prop = node_prop(neighbor);
+          if (not neighbor_prop.IsFrozen() and
+              FullEdgeSize(find(vertex), it)==1 + neighbor_prop.size()) {
+              FreezeVertex(neighbor);
+              new_frozen.insert(neighbor);
+          }
+      }
+    };
+
+    while (not prev_frozen.empty()) {
+        for (const RRVertexType &vertex : prev_frozen) {
+            auto[in_nbr_begin, in_nbr_end] = in_neighbors(vertex);
+            auto[out_nbr_begin, out_nbr_end] = out_neighbors(vertex);
+            upd_new_frozen(vertex, node_prop(vertex), in_nbr_begin, in_nbr_end);
+            upd_new_frozen(vertex,
+                           node_prop(vertex),
+                           out_nbr_begin,
+                           out_nbr_end);
+        }
+        prev_frozen = std::move(new_frozen);
+    }
+}
+
+void MultiplexDBG::FreezeUnpairedVertices() {
+    for (const RRVertexType &vertex : *this) {
+        RRVertexProperty &vertex_prop = node_prop(vertex);
+        if (vertex_prop.IsFrozen()) {
+            continue;
+        }
+
+        auto[in_edges, out_edges] = GetNeighborEdgesIndexes(vertex);
```

Comment:
Indexes->Indices

Reply:
Seems like both can be used with indexes a bit more common in American English:
- https://www.grammar.com/indices_vs._indexes
- https://grammarist.com/usage/indexes-indices/
- https://stackoverflow.com/questions/1378781/proper-terminology-should-i-say-indexes-or-indices

Please also note what google ngram viewer is showing — indexes looks more common:
https://books.google.com/ngrams/graph?content=indexes%2Cindeces&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Cindexes%3B%2Cc0%3B.t1%3B%2Cindeces%3B%2Cc0

There are multiple places in the code where I use indexes, so unless there is a strong incentive for a change, I would prefer to skip that for now.